File: 560-subarray-sum-equals-k/subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
       unordered_map<int,int>mp;
       mp[0]=1;
       int count =0;
       int sum =0;
       for(int i=0;i<nums.size();i++){
            // find for target -curr_elmenet
            sum+=nums[i];
            int element= sum-k;
            if(mp.find(element)!=mp.end()){
                    count+=mp[element];
            }
            mp[sum]++;
       } 
       return count;
    }
};

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var subarraySum = function(nums, k) {
       let mp=new Map();
       let count=0;
       let sum=0;
       mp.set(0,1);
       for(let i =0;i<nums.length;i++){
            sum+=nums[i];
            
                if (mp.has(sum - k)) {
            count += mp.get(sum - k);
        }

        mp.set(sum, (mp.get(sum) || 0) + 1);
       }
       return count;
};

//sliding window approach
